processing/src/main/java/org/apache/druid/frame/write/RowBasedFrameWriter.java

```diff
@@ -120,14 +119,8 @@ public boolean addSelection()
       return false;
     }
 
-    try {
-      if (!writeData()) {
-        return false;
-      }
-    }
-    catch (Exception e) {
-      Throwables.propagateIfInstanceOf(e, ParseException.class);
-      throw new ParseException("", e, "Unable to add the row to the frame. Type conversion might be required.");
```

Comment:
This shouldn't break any other exception handling since we don't rely on this anywhere in the callers.
This was solely added for the purpose of catching InvalidNullByteException earlier, which was incorrect since frame writer exceptions can be due to multiple purposes, however, we were wrapping them unconditionally into ParseException. This PR reverts the original change that was made because it's incorrect to do so and we should selectively wrap those exceptions.